internal/net/ssl/ssl.go

```diff
@@ -178,6 +185,107 @@ func AddOrUpdateCertAndKey(name string, cert, key, ca []byte,
 		PemSHA:      file.SHA1(pemFileName),
 		CN:          cn.List(),
 		ExpireTime:  pemCert.NotAfter,
+		PemCertKey:  string(pemCerts),
+	}
+
+	return s, nil
+}
+
+// CreateSSLCert creates an SSLCert and avoids writing on disk unless there is a CA
```

Comment:
> avoids writing on disk unless there is a CA

why is this the case?

Reply:
When there is a CA, `CAFileName` gets created and in `nginx.tmpl` `ssl_client_certificate` is set to this. It doesn't look like OpenResty can set `ssl_client_certificate` dynamically so we have to write this on disk.